src/home-page/__tests__/NavBar.test.js

```diff
@@ -0,0 +1,32 @@
+import { render } from '@testing-library/react';
+import { fireEvent, screen } from '@testing-library/dom';
+import React from 'react';
+import NavBar from '../NavBar';
+import { AuthContext } from '../../context/AuthProvider';
+
+describe('tests for navbar', async () => {
+  const value = {
+    username: 'razvan',
+    userRole: 3,
+    setUsername: jest.fn(),
+    setUserRole: jest.fn(),
+  };
+
+  beforeEach(async () => {
+    render(
+      <AuthContext.Provider value={value}>
+        <NavBar />
+      </AuthContext.Provider>,
+    );
+  });
+
+  it('should display the user name', async () => {
+    expect(screen.getByText('Hello, razvan!')).toBeInTheDocument();
+  });
+
+  it('should setUsername to null if Logout button is pressed', async () => {
```

Comment:
There is a mismatch between the description of the test and what it actually tests.

Reply:
yes, i actually forgot to change the name
fixed